devnet-sdk/testing/systest/systest.go

```diff
@@ -28,22 +87,40 @@ func SystemTest(t BasicT, f SystemTestFunc, validators ...PreconditionValidator)
 	for _, validator := range validators {
 		ctx, err := validator(wt, sys)
 		if err != nil {
-			t.Skipf("validator failed: %v", err)
+			h.handlePreconditionError(t, err)
 		}
 		wt = wt.WithContext(ctx)
 	}
 
 	f(wt, sys)
 }
 
-type InteropSystemTestFunc func(t T, sys system.InteropSystem)
-
-func InteropSystemTest(t BasicT, f InteropSystemTestFunc, validators ...PreconditionValidator) {
-	SystemTest(t, func(t T, sys system.System) {
+func (h *basicSystemTestHelper) InteropSystemTest(t BasicT, f InteropSystemTestFunc, validators ...PreconditionValidator) {
+	t.Helper()
+	h.SystemTest(t, func(t T, sys system.System) {
 		if sys, ok := sys.(system.InteropSystem); ok {
 			f(t, sys)
 		} else {
-			t.Skipf("interop test requested, but system is not an interop system")
+			h.handlePreconditionError(t, fmt.Errorf("interop test requested, but system is not an interop system"))
 		}
 	}, validators...)
 }
+
+// newBasicSystemTestHelper creates a new basicSystemTestHelper using environment variables
+func newBasicSystemTestHelper(env envGetter) *basicSystemTestHelper {
+	val := env.Getenv(EnvVarExpectPreconditionsMet)
+	expectPreconditionsMet := val != "" && val != "0"
```

Comment:
Just a thought; might any set (non empty) value suffice to indicate here? That may simplify things a touch.

Reply:
That's fair. I guess I better provide some library helper to setup an environment programmatically. I'll do that in a followup.

Actually let me plug in ParseBool here, that'll be more consistent